### ISSUE TITLE
SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:726)

### ISSUE THREAD

**Issue by josh-paul** [kubernetes-client/python#550]
Getting the above error when using the admin.conf from the apiserver or service account automounted credentials in a pod. Have tried on python 2.7.15 and python 3.6.5. Looking for any solution to this. The code used to work fine until recently. Kubernetes version 1.10.4 installed via kubeadm. The same credentials work with kubectl to access the kube.

`from kubernetes import client, config
config.load_incluster_config()
v1 = client.CoreV1Api()`

`from kubernetes import client, config
config.load_kube_config(config_file)
v1 = client.CoreV1Api()`

`root@5f2529864952:/# pip list
Package           Version  
----------------- ---------
boto3             1.7.40   
botocore          1.10.40  
cachetools        2.1.0    
certifi           2018.4.16
chardet           3.0.4    
docutils          0.14     
dotted-dict       1.0.8    
futures           3.2.0    
google-auth       1.5.0    
idna              2.7      
ipaddress         1.0.22   
jmespath          0.9.3    
kubernetes        6.0.0    
oauthlib          2.1.0    
pip               10.0.1   
pyasn1            0.4.3    
pyasn1-modules    0.2.1    
python-dateutil   2.7.3    
PyYAML            3.12     
requests          2.19.1   
requests-oauthlib 1.0.0    
rsa               3.4.2    
s3transfer        0.1.13   
setuptools        39.2.0   
six               1.11.0   
urllib3           1.23     
virtualenv        16.0.0   
websocket-client  0.48.0   
wheel             0.31.1`

**Comment by josh-paul** [comment]
This is due to how if a ca cert is passed in, Python does not attempt any further look up in certifi.  K8s passes back the intermediate ca cert only in the credentials. Since the full chain is not there, python fails verification. My work around is to remove the certificate-authority-data from the kube config file. I have not found a work around for using automounted service account credentials.

**Comment by pengisgood** [comment]
I got the same issue on Mac OSX 10.13.6 with the logs:
```
kubectl version
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.2", GitCommit:"bb9ffb1654d4a729bb4cec18ff088eacc153c239", GitTreeState:"clean", BuildDate:"2018-08-08T16:31:10Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:00:59Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

python -V
Python 3.5.0

pip list
Package           Version
----------------- ---------
adal              1.0.2
asn1crypto        0.24.0
cachetools        2.1.0
certifi           2018.8.24
cffi              1.11.5
chardet           3.0.4
cryptography      2.3.1
google-auth       1.5.1
idna              2.7
kubernetes        7.0.0
ndg-httpsclient   0.5.1
oauthlib          2.1.0
pip               18.0
pyasn1            0.4.4
pyasn1-modules    0.2.2
pycparser         2.18
PyJWT             1.6.4
pyOpenSSL         18.0.0
python-dateutil   2.7.3
PyYAML            3.13
requests          2.19.1
requests-oauthlib 1.0.0
rsa               3.4.2
setuptools        40.2.0
six               1.11.0
urllib3           1.23
websocket-client  0.51.0
```

**Comment by hightoxicity** [comment]
> This is due to how if a ca cert is passed in, Python does not attempt any further look up in certifi. K8s passes back the intermediate ca cert only in the credentials. Since the full chain is not there, python fails verification. My work around is to remove the certificate-authority-data from the kube config file. I have not found a work around for using automounted service account credentials.

I managed to pass the full cert bundle (my own external to kube root ca + kube intermed ca generated under the first one), that is now contained into /var/run/secrets/kubernetes.io/serviceaccount/ca.crt of my running pods.
I did it thanks to the --root-ca-file parameter of the kube-controller-manager (https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/)

But after having set it, calico seems to not be able to talk to apiserver, and kube can no more prepare  the network namespace of new containers to run...
I hope I missed something and that we will be able to find a trick to be able to use this python cli with sa token.
From my side, the failure come from the load_incluster_config of the ceph manager, this pod is created by rook operator.
